### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: libevmjit/JIT.cpp [hunk 1]
```diff
@@ -189,10 +189,19 @@ static int64_t call_v2(
 	// Handle output. It can contain data from RETURN or REVERT opcodes.
 	auto size = std::min(_outputSize, result.output_size);
 	std::copy(result.output_data, result.output_data + size, _outputData);
-	jit.returnBuffer = {result.output_data, result.output_data + result.output_size};
 
-	*o_bufData = jit.returnBuffer.data();
-	*o_bufSize = jit.returnBuffer.size();
+	if (_kind == EVM_CREATE && result.status_code == EVM_SUCCESS)
```
Comment (chfast) [inline libevmjit/JIT.cpp]: @winsvega @pirapira Can you review this part?

File: libevmjit/JIT.cpp [hunk 2]
```diff
@@ -531,5 +533,39 @@ JITImpl::JITImpl():
...
+	createEngine();
+}
+
+void JITImpl::checkMemorySize()
+{
+	constexpr size_t memoryLimit = 100 * 1024 * 1024;
+
+	if (m_memoryMgr->totalMemorySize() > memoryLimit)
+	{
+		std::cerr << "EVMJIT reset!\n";
```
Comment (axic) [inline libevmjit/JIT.cpp]: Is this a cache clear or a full reset?

File: libevmjit/JIT.cpp (hunk 1 above)
Comment (pirapira) [reply libevmjit/JIT.cpp]: Looks good.

File: libevmjit/JIT.cpp (hunk 2 above)
Comment (chfast) [reply libevmjit/JIT.cpp]: This log should be protected by `-st` flag.

Anyway, in legacy MCJIT API from LLVM there is no way to remove compiled module. The only way to release memory is to delete MCJIT instance and create a new one.

File: libevmjit/JIT.cpp (hunk 1 above)
Comment (chfast) [reply libevmjit/JIT.cpp]: I replaced this commit with better fix that handles also the case of CALL exiting early doe to insufficient balance.

**Comment by chfast** [comment]
Please do another review. Thanks.

**Comment by pirapira** [comment]
Looks good.  Yes, it's safer to reset the buffer every time before a call or a create.

**Pull request merged**